core/services/ocr2/plugins/ccip/internal/ccipdb/price_service.go

```diff
@@ -242,153 +257,182 @@ func (p *priceService) runCleanup(ctx context.Context) error {
 	// We can afford to have non-performant unlocks here that is simple and safe.
 	p.dynamicConfigMu.RLock()
 	defer p.dynamicConfigMu.RUnlock()
 
 	// There may be a period of time between service is started and dynamic config is updated
-	if p.gasPriceEstimator == nil || p.destPriceRegistryReader == nil {
-		p.lggr.Info("Skipping price update due to gasPriceEstimator and/or destPriceRegistry not ready")
+	if p.gasPriceEstimator == nil {
+		p.lggr.Info("Skipping gas price update due to gasPriceEstimator not ready")
+		return nil
+	}
+
+	sourceGasPriceUSD, err := p.observeGasPriceUpdates(ctx, p.lggr)
+	if err != nil {
+		return fmt.Errorf("failed to observe gas price updates: %w", err)
+	}
+
+	err = p.writeGasPricesToDB(ctx, sourceGasPriceUSD)
+	if err != nil {
+		return fmt.Errorf("failed to write gas prices to db: %w", err)
+	}
+
+	return nil
+}
+
+func (p *priceService) runTokenPriceUpdate(ctx context.Context) error {
+	// Protect against concurrent updates of `tokenPriceEstimator` and `destPriceRegistryReader`
+	// Price updates happen infrequently - once every `tokenPriceUpdateInterval` seconds.
+	p.dynamicConfigMu.RLock()
+	defer p.dynamicConfigMu.RUnlock()
+
+	// There may be a period of time between service is started and dynamic config is updated
+	if p.destPriceRegistryReader == nil {
+		p.lggr.Info("Skipping token price update due to destPriceRegistry not ready")
 		return nil
 	}
 
-	sourceGasPriceUSD, tokenPricesUSD, err := p.observePriceUpdates(ctx, p.lggr)
+	tokenPricesUSD, err := p.observeTokenPriceUpdates(ctx, p.lggr)
 	if err != nil {
-		return fmt.Errorf("failed to observe price updates: %w", err)
+		return fmt.Errorf("failed to observe token price updates: %w", err)
 	}
 
-	err = p.writePricesToDB(ctx, sourceGasPriceUSD, tokenPricesUSD)
+	err = p.writeTokenPricesToDB(ctx, tokenPricesUSD)
 	if err != nil {
-		return fmt.Errorf("failed to write prices to db: %w", err)
+		return fmt.Errorf("failed to write token prices to db: %w", err)
 	}
 
 	return nil
 }
 
-func (p *priceService) observePriceUpdates(
+func (p *priceService) observeGasPriceUpdates(
 	ctx context.Context,
 	lggr logger.Logger,
-) (sourceGasPriceUSD *big.Int, tokenPricesUSD map[cciptypes.Address]*big.Int, err error) {
-	if p.gasPriceEstimator == nil || p.destPriceRegistryReader == nil {
-		return nil, nil, fmt.Errorf("gasPriceEstimator and/or destPriceRegistry is not set yet")
+) (sourceGasPriceUSD *big.Int, err error) {
+	if p.gasPriceEstimator == nil {
+		return nil, fmt.Errorf("gasPriceEstimator is not set yet")
 	}
 
-	sortedLaneTokens, filteredLaneTokens, err := ccipcommon.GetFilteredSortedLaneTokens(ctx, p.offRampReader, p.destPriceRegistryReader, p.priceGetter)
+	// Include wrapped native to identify the source native USD price, notice USD is in 1e18 scale, i.e. $1 = 1e18
+	rawTokenPricesUSD, err := p.priceGetter.TokenPricesUSD(ctx, []cciptypes.Address{p.sourceNative})
+	if err != nil {
+		return nil, err
```

Comment:
Would it be helpful to add more context here? Since this call is being made to calculate gas prices, e.g

```suggestion
		return nil, fmt.Errorf("failed to fetch wrapped native/usd price (%s) when observing gas prices: %w", p.sourceNative, err)
```